packages/testing/src/consensus_testing/test_fixtures/fork_choice.py

```diff
@@ -125,6 +161,26 @@ def make_fixture(self) -> "ForkChoiceTest":
         # Ensure anchor_state and anchor_block are set
         assert self.anchor_state is not None, "anchor_state must be set before make_fixture"
         assert self.anchor_block is not None, "anchor_block must be set before make_fixture"
+        assert self.max_slot is not None, "max_slot must be set before make_fixture"
+
+        # Initialize XMSS key manager with max_slot for key generation
+        # Avoid adding excess buffer as it slows down test execution
+        key_manager = XmssKeyManager(max_slot=self.max_slot)
+
+        # Update validator pubkeys to match key_manager's generated keys
+        updated_validators = []
+        for i, validator in enumerate(self.anchor_state.validators):
+            pubkey, _ = key_manager.create_key_pair(ValidatorIndex(i))
```

Comment:
Nit: I'm not sure if it's just me but I think the name implies it's pure function (stateless, no side effects) while behind the scene it's actually storing the key pair in the manager before returning. Can we make this more visible? E.g.

```suggestion
            key_manager.create_key_pair(ValidatorIndex(i))
            pubkey, _ = key_manager.get_key_pair(ValidatorIndex(i))
```

or


```suggestion
            pubkey, _ = key_manager.create_and_store_key_pair(ValidatorIndex(i))
```